chef/data_bags/crowbar/migrate/neutron/030_seperate_ml2_drivers.rb

```diff
@@ -0,0 +1,59 @@
+def upgrade ta, td, a, d
+  if a['networking_mode']
+    a['ml2_type_drivers'] = a['networking_mode']
+  else
+    a['ml2_type_drivers'] = ta['ml2_type_drivers']
+  end
+
+  if a['networking_plugin'] == 'linuxbridge'
+    a['networking_plugin'] = 'ml2'
+    a['ml2_mechanism_drivers'] = 'linuxbridge'
+  elsif a['networking_plugin'] == 'openvswitch'
+    a['networking_plugin'] = 'ml2'
+    a['ml2_mechanism_drivers'] = 'openvswitch'
+  elsif a['networking_plugin'] == 'cisco'
+    a['networking_plugin'] = 'ml2'
+    a['ml2_mechanism_drivers'] = 'cisco'
+  elsif a['networking_plugin'] == 'vmware'
+    a['networking_plugin'] = 'vmware'
+  else
+    a['networking_plugin'] = ta['networking_plugin']
+    a['ml2_mechanism_drivers'] = ta['ml2_mechanism_drivers']
+  end
+
+  a.delete('networking_mode')
+
+  return a, d
+end
+
+def downgrade ta, td, a, d
+  if a.has_key? 'ml2_type_drivers'
+    if a['ml2_type_drivers'].include? 'vlan'
+      a['networking_mode'] = 'vlan'
+    elsif a['ml2_type_drivers'].include? 'gre'
+      a['networking_mode'] = 'gre'
+    else
+      a['networking_mode'] = ta['networking_mode']
+    end
+  else
+    a['networking_mode'] = ta['networking_mode']
+  end
+
+  if a['networking_plugin'] == 'ml2'
+    if a['ml2_mechanism_drivers'].include? 'linuxbridge'
+      a['networking_plugin'] = 'linuxbridge'
+    elsif a['ml2_mechanism_drivers'].include? 'openvswitch'
+      a['networking_plugin'] = 'openvswitch'
+    elsif a['ml2_mechanism_drivers'].include? 'cisco'
+      a['networking_plugin'] = 'cisco'
+    elsif a['ml2_mechanism_drivers'].include? 'cisco'
+      a['networking_plugin'] = 'cisco'
```

Comment:
> Duplicated lines.

Done in https://github.com/toabctl/barclamp-neutron/commit/bb6f0757b4155c9fb84e87254977e84028a66cbc